lib/machine/configure.ts

```diff
@@ -132,7 +154,20 @@ export function configure<T extends SdmContext = PushListenerInvocation>(
                         configuration: cfgToUse,
                     });
 
-                const configured = await configurer(sdm);
+                // Decorate the createGoals method onto the SDM
+                (sdm as any).createGoals = async (creator: GoalCreator<any>, configurers: GoalConfigurer<any> | Array<GoalConfigurer<any>>) => {
```

Comment:
It seems kind of sneaky to add createGoals to the SDM and then take it away.

Reply:
It certainly is sneaky but it is only there for the duration of the `configure` method with specifies it in the method signature. So people can't expect that method to exist later. Ok?